Core/include/Acts/Seeding/SeedFinderUtils.ipp

```diff
@@ -92,10 +92,21 @@ void transformCoordinates(
     l.Er = ((varianceZM + sp->varianceZ()) +
             (cot_theta * cot_theta) * (varianceRM + sp->varianceR())) *
            iDeltaR2;
+
+    l.x = sp->x();
+    l.y = sp->y();
+    l.z = sp->z();
+    l.r = sp->radius();
+
     linCircleVec.push_back(l);
   }
   // sort the SP in order of cotTheta
   if (enableCutsForSortedSP) {
+    std::sort(vec.begin(), vec.end(),
+              [](const InternalSpacePoint<external_spacepoint_t>* a,
+                 const InternalSpacePoint<external_spacepoint_t>* b) -> bool {
+                return (a->cotTheta() < b->cotTheta());
+              });
     std::sort(linCircleVec.begin(), linCircleVec.end(),
```

Comment:
Do we need to sort both the space point vector and the `LinCircleVec`? Can you check if the `LinCircleVec` gets sorted automatically if you sort the space point vector before looping on its content?

Reply:
I am filling the space points with `cotTheta` during the loop, and using these values to sort the space point vector. In order to sort the space point vector before the loop I will need to calculate the values of `cotTheta` before the loop, as well. Or maybe there is a better way to sort these vectors?